python/ray/serve/backend_worker.py

```diff
@@ -251,6 +253,31 @@ def get_runner_method(self, request_item: Query) -> Callable:
             return self.callable
         return getattr(self.callable, method_name)
 
+    async def ensure_serializable_response(self, response: Any) -> Any:
+        if isinstance(response, starlette.responses.StreamingResponse):
+            # response contains a generator/iterator which is not serializable.
+            # Exhaust the generator/iterator and store the results in a buffer.
+            body_buffer = []
+
+            async def mock_send(message):
+                if (message["type"] == "http.response.body"):
+                    body_buffer.append(message["body"])
```

Comment:
I think the following will convey the idea that message can only be one type better.
```suggestion
                assert message["type"] == "http.response.body"
                body_buffer.append(message["body"])
```

Reply:
Turns out at the beginning it sends a `"http.response.start"`.  I've changed the code to this:
```python
            async def mock_send(message):
                assert message["type"] in {
                    "http.response.start", "http.response.body"
                }
                if (message["type"] == "http.response.body"):
                    body_buffer.append(message["body"])
```